addons/mail/models/mail_channel_partner.py

```diff
@@ -29,6 +28,17 @@ class ChannelPartner(models.Model):
     last_interest_dt = fields.Datetime("Last Interest", default=fields.Datetime.now, help="Contains the date and time of the last interesting event that happened in this channel for this partner. This includes: creating, joining, pinning, and new message posted.")
     rtc_inviting_session_id = fields.Many2one('mail.channel.rtc.session', string='Ringing session')
 
+    @api.depends('guest_id', 'partner_id')
+    def name_get(self):
+        res = []
+        for record in self:
+            if record.guest_id:
+                name = record.guest_id.name
+            else:
+                name = record.partner_id.name
+            res.append((record.id, name))
+        return res
+
```

Comment:
```suggestion
    @api.depends('guest_id.name', 'partner_id.name')
    def name_get(self):
        return [(record.id, record.partner_id.name or record.guest_id.name) for record in self]

```

Reply:
Actually don't even need the depends line, `name_get` is just a method not a compute.

Reply:
If you want display name to adjust dynamically then you need to override the field or compute of display name to add the depends.

Reply:
@seb-odoo I see, I added the depends because I saw that some (although few) name_get had it.